### ISSUE TITLE
[FEATURE] Consider simplifying use of bokeh_server_page in tests

### ISSUE THREAD

**Issue by ltalirz** [bokeh/bokeh#10479]
As mentioned by @bryevdv [here](https://github.com/bokeh/bokeh/pull/10476#discussion_r485218670)

> The previous tests do a further round trip update and check the value new from a data source update via a `CustomAction` (on a plot) In hindsight all the tests could probably be simplified to be more like this one.

Example of the previous tests:
https://github.com/bokeh/bokeh/blob/658f80d22fe9900a08435bb51cdf59b0900028a6/tests/integration/widgets/test_autocomplete_input.py#L349-L359

Example of alternative approach:
```python
    def test_server_strict(self, bokeh_server_page) -> None:
        """Test effect of 'strict=False' without explicit callback."""
        text_input = AutocompleteInput(css_classes=["foo"], completions = ["aAaBbb", "aAaBbB"], strict=False)

        def add_autocomplete(doc):
            # note: for some reason, bokeh_server_page requires a 'canvas' in the document
            plot = Plot()
            doc.add_root(column(text_input,plot))

        page = bokeh_server_page(add_autocomplete)

        el = page.driver.find_element_by_css_selector('.foo input')
        text = "not in completions"
        enter_text_in_element(page.driver, el, text, click=1, enter=True)

        assert text_input.value == text
```

**Comment by bryevdv** [comment]
My one concern with simplifying the tests in this way is that it is definitely a pattern of usage that is *only* viable in tests. Specifically, in normal usage, it is absolutely not possible to create Bokeh objects *outside* the `modify_doc(doc)` function, because then all new sessions would share that same instance of the Bokeh model, and that is prohibited. It works by coincidence in the test because the test only ever results in a single session being created. 

Anyway to summarize pros/cons:

### pros

* Makes test code considerably simpler

### cons

* Illustrates a "bad" usage pattern that should not be emulated anywhere else
* Would break if any test ever created multiple sessions instead of just one

**Comment by mattpap** [comment]
This was rendered irrelevant since selenium based integration tests were removed in PR #13992.
